### PR TITLE
Make parity.js usable by Node and Browser

### LINKED ISSUE #3471: Parity.js relying on global `fetch`
```
const Api = require('@parity/parity.js').Api;
const transport = new Api.Transport.Http('http://localhost:8545');
const api = new Api(transport);

api.eth
  .coinbase()
  .then((coinbase) => {
    console.log('The coinbase is', coinbase);
  });

=======================================================

/home/kyroy/node_modules/@parity/parity.js/library.js:3
[...]

ReferenceError: fetch is not defined
    at e.value (/home/kyroy/node_modules/@parity/parity.js/library.js:3:15183)
    at e.r._pollConnection (/home/kyroy/node_modules/@parity/parity.js/library.js:3:14669)
    at new e (/home/kyroy/node_modules/@parity/parity.js/library.js:3:14760)
    at Object.<anonymous> (/home/kyroy/github/offchain-computation-verification/contracts/mm_single_client_parity.js:2:17)
    at Module._compile (module.js:573:32)
    at Object.Module._extensions..js (module.js:582:10)
    at Module.load (module.js:490:32)
    at tryModuleLoad (module.js:449:12)
    at Function.Module._load (module.js:441:3)
    at Module.runMain (module.js:607:10)
```

Should probably use some kind of https://github.com/matthew-andrews/isomorphic-fetch instead.

### PR DESCRIPTION
**Pull request by ngotchac**
Fixes #3471

```javascript
const Api = require('/path/to/parity/js/.build/parity.js').Api;
const transport = new Api.Transport.Http('http://localhost:8545');
const api = new Api(transport);

api.eth
  .blockNumber()
  .then((block) => {
    console.log('At block', block.toFormat());
    process.exit(0);
  })
  .catch((e) => {
    console.error(e);
    process.exit(1);
  });
```

### CODE REVIEW

**Comment by tomusdrw** [comment]
Why not just use `const fetch = require('isomorphic-fetch');` instead of relying on global variable to be present?

**Comment by jacogr** [comment]
Agreed with @tomusdrw here - just include isomorphic-fetch as part of the entry point (instead of whatwg-fetch) and no other hoops to jump through. (Just webpack lib build needs to include it, which it will)

**Comment by ngotchac** [comment]
So I'm not sure how this library is distributed on NPM... ? I had to do this trick because I thought the built JS file had to work in Node and in a browser

**Comment by tomusdrw** [comment]
Hmm, so I thought that`isomorphic-fetch` is actually taking care of that (choosing right polyfill depending on the environment).
So my understanding was that we should not just "append that to the build", but rather alter the code using `fetch` to import it explicilty.

**Comment by ngotchac** [comment]
Well `isomorphic-fetch` loads the right module when in a Node environment, but when Webpack builds the library, it will try to get all the necessary code for the Node side of `isomorphic-fetch` anyway.

The real solution is to have a proper repo, with a build for browser and a build for Node

**Comment by ngotchac** [comment]
@jacogr @tomusdrw This should be better now. `node-fetch` is a dependency of the library, and the JS `parity` library only bundles `whatwg-fetch`.

**Comment by jacogr** [comment]
Ok, as discussed, seems it is the best approach.

I just want to test the library as built in both a Node & Browser environment before giving the ok. (Don't want to go cowboy and have a broken module on NPM. Will do so first thing in the morning.)

**Comment by gavofyork** [comment]
ci failing?

**Comment by jacogr** [comment]
@ngotchac Just added the in-progress label since there are issues you are working on in here.

**Comment by ngotchac** [comment]
@jacogr Ok it should be fine now.

You can try it with `npm run test:npm` for Node integration. In the browser, the `window.Parity` object is created with `Api` and `Abi`.

File: js/webpack.npm.js [hunk 1]
```diff
@@ -24,13 +24,23 @@ const isProd = ENV === 'production';
 
 module.exports = {
   context: path.join(__dirname, './src'),
+  target: 'node',
   entry: 'library.js',
   output: {
     path: path.join(__dirname, '.npmjs'),
     filename: 'library.js',
-    libraryTarget: 'commonjs'
+    library: 'Parity',
```
Comment (jacogr) [inline js/webpack.npm.js]: s/Parity/parity/, need this to work -

```
const { api } = window.parity;
```

**Pull request merged**